modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusThreadPoolExecutor.java

```diff
@@ -15,11 +18,14 @@ private StatusThreadPoolExecutor() {
         mQueue = new LinkedBlockingQueue<>();
 
         mThreadPool = new ThreadPoolExecutor(
-            NUMBER_OF_CORES,
-            NUMBER_OF_CORES,
+            100 * NUMBER_OF_CORES,
```

Comment:
Shouldn't there be a constant `THREADS_TO_CORES_RATIO` to make it clearer?

Reply:
@reutsharabani @PombeirP  I'll update with `THREADS_TO_CORES_RATIO` later today. My instinct about `Integer.MAX_VALUE` is it would allow a runaway/misbehaving client to have a worse impact on the rest of the OS than if threads are constrained, don't you reckon? If the thread count gets really big the phone is spending all its time context switching and the threads just make things worse...

Reply:
@pacamara I completely agree. The thread count should correspond with the system resources. No hardware I know can run `Integer.MAX_VALUE` threads without locking the CPU(s) forever and exhausting the memory (for the per-thread stacks - especially in Java).

I think the core count is a good indicator of the system's capabilities and therefore so the thread count can be reasonable determined by it.

Reply:
Yeah, I agree that `Integer.MAX_VALUE` is not the right solution (that would be negating the advantages of having moved from creating threads manually to using a thread pool), but I'm also not sure that having 800 threads in a 8-core phone is feasible. Ideally we'd know when we screw up for some reason and create some new code that blocks threads for a long time, causing the thread pool to be fully utilized for a long time.